ocaml/xapi-cli-server/records.ml

```diff
@@ -5242,3 +5242,119 @@ let vtpm_record rpc session_id vtpm =
           ()
       ]
   }
+
+let tracing_record rpc session_id tracing =
+  let _ref = ref tracing in
+  let empty_record =
+    ToGet (fun () -> Client.Tracing.get_record ~rpc ~session_id ~self:!_ref)
+  in
+  let record = ref empty_record in
+  let x () = lzy_get record in
+  {
+    setref=
+      (fun r ->
+        _ref := r ;
+        record := empty_record
+      )
+  ; setrefrec=
+      (fun (a, b) ->
+        _ref := a ;
+        record := Got b
+      )
+  ; record= x
+  ; getref= (fun () -> !_ref)
+  ; fields=
+      [
+        make_field ~name:"uuid" ~get:(fun () -> (x ()).API.tracing_uuid) ()
+      ; make_field ~name:"hosts"
+          ~get:(fun () ->
+            List.map get_uuid_from_ref (x ()).API.tracing_hosts
+            |> String.concat ","
+            |> fun s -> if s = "" then "all" else ""
+          )
+          ()
+      ; make_field ~name:"name_label"
+          ~get:(fun () -> (x ()).API.tracing_name_label)
+          ()
+      ; make_field ~name:"tags"
```

Comment:
This is a map, I recommend to follow how other maps are implemented in here. See the field type to see the interesting fields to define. These are get_map, add_to_map, remove_from_map, set_in_map, set_map and clear_map instead of defining a custom set, but I doubt all of them are needed

Reply:
This is for tags, which is modelled as a map